espflash/src/cli/serial.rs

```diff
@@ -22,9 +22,13 @@ pub fn get_serial_port_info(
     let ports = detect_usb_serial_ports().unwrap_or_default();
 
     if let Some(serial) = &matches.serial {
```

Comment:
Would you mind adding a comment explaining this please? There are a couple things going on here and it isn't immediately clear to others what the motivation for this is:

- The call to `canonicalize()` was originally added to resolve #177
- This then causes issues on Windows, so we need to handle Windows/Posix differently